addons/website_event_booth_exhibitor/models/event_booth_category.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class EventBoothCategory(models.Model):
+    _inherit = 'event.booth.category'
+
+    @api.model
+    def _get_exhibitor_type(self):
+        return self.env['event.sponsor']._fields['exhibitor_type'].selection
+
+    use_sponsor = fields.Boolean(string='Create Sponsor/Exhibitor', help="If set, when booking a booth a sponsor will be created for the user")
+    sponsor_type_id = fields.Many2one('event.sponsor.type', string='Sponsor Level')
```

Comment:
We could do a bit like on sponsor model: if people choose "use sponsor", choose a sponsor type to have something by default in the field (``self.env['event.sponsor.type'].search([], order="sequence desc", limit=1).id``) ? Maybe either an onchange, either a computed editable field. As it would be something UI related, maybe an onchange ?